File: codesmith/common/naming.go
// Names for some CloudFormation custom resources.
package common

import (
	"github.com/pkg/errors"
	"regexp"
)

// SSM Parameters
//
// Some of the custom CloudFormation resources create by the forge are implemented as SSM parameters in the parameter
// store.
const SsmParameterPrefix = "/codesmith-forge"

// Dns Certificate SNS Message memory
func DnsCertificeSnsMessageIdParameterName(stackArn string, logicalResourceID string) (string, error) {
	stackId, err := ExtractStackId(stackArn)
	if err != nil {
		return "", err
	}
	return SsmParameterPrefix + "/DnsCertificateSnsMessageId/" + stackId + "/" + logicalResourceID, nil
}

var stackIdRegExp = regexp.MustCompile("^arn:.*:cloudformation:.*:.*:stack/(.*)$")

func ExtractStackId(stackArn string) (string, error) {
	submatch := stackIdRegExp.FindStringSubmatch(stackArn)
	if len(submatch) != 2 {
		return "", errors.Errorf("Could not extract stackId from stack arn %s", stackArn)
	}
	return submatch[1], nil
}

var stackNameRegExp = regexp.MustCompile("^arn:.*:cloudformation:.*:.*:stack/(.*?)/.*$")

func ExtractStackName(stackArn string) (string, error) {
	submatch := stackNameRegExp.FindStringSubmatch(stackArn)
	if len(submatch) != 2 {
		return "", errors.Errorf("could not extract stackName for stack arn %s", stackArn)
	}
	return submatch[1], nil
}

// Sequence naming
func SequenceParameterName(sequenceName string) string {
	return SsmParameterPrefix + "/Sequence" + sequenceName
}

// CogCondPreAuth naming
func CogCondPreAuthParameterName(userPoolId string, userPoolClientId string) string {
	return SsmParameterPrefix + "/CogCondPreAuth/" + userPoolId + "/" + userPoolClientId
}

File: codesmith/common/naming.py
import re

# SSM Parameters
#
# Some of the custom CloudFormation resources create by the forge are implemented
# as SSM parameters in the parameter store.
SSM_PARAMETER_PREFIX = '/codesmith-forge'


# CogCondPreAuth naming
def cog_cond_pre_auth_parameter_name(user_pool_id: str, user_pool_client_id: str) -> str:
    return '{0}/CogCondPreAuth/{1}/{2}'.format(SSM_PARAMETER_PREFIX, user_pool_id, user_pool_client_id)


# Sequence naming
def sequence_parameter_name(sequence_name: str) -> str:
    return '{0}/Sequence{1}'.format(SSM_PARAMETER_PREFIX, sequence_name)


# Dns Certificate SNS Message memory
def dns_certificate_sns_message_id_parameter_name(stack_arn, logical_resource_id):
    stack_id = extract_stack_id(stack_arn)
    return '{0}/DnsCertificateSnsMessageId/{1}/{2}'.format(SSM_PARAMETER_PREFIX, stack_id, logical_resource_id)


# S3ReleaseCleanup store
def s3_release_cleanup_parameter_name(stack_arn, logical_resource_id):
    stack_id = extract_stack_id(stack_arn)
    return '{0}/S3ReleaseCleanup/{1}/{2}'.format(SSM_PARAMETER_PREFIX, stack_id, logical_resource_id)


STACK_ID_REGEX = re.compile('^arn:.*:cloudformation:.*:.*:stack/(.*)')


def extract_stack_id(stack_arn):
    return STACK_ID_REGEX.match(stack_arn).group(1)
